src/webviews/apps/shared/app.ts

```diff
@@ -0,0 +1,114 @@
+import { provide } from '@lit/context';
+import { html } from 'lit';
+import { property } from 'lit/decorators.js';
+import type { CustomEditorIds, WebviewIds, WebviewViewIds } from '../../../constants.views';
+import type { Deferrable } from '../../../system/function';
+import { debounce } from '../../../system/function';
+import type { WebviewFocusChangedParams } from '../../protocol';
+import { DidChangeWebviewFocusNotification, WebviewFocusChangedCommand, WebviewReadyCommand } from '../../protocol';
+import { GlElement } from './components/element';
+import { ipcContext, loggerContext, LoggerContext } from './context';
+import type { Disposable } from './events';
+import { HostIpc } from './ipc';
+
+export abstract class GlApp<
+	State extends { webviewId: CustomEditorIds | WebviewIds | WebviewViewIds; timestamp: number } = {
+		webviewId: CustomEditorIds | WebviewIds | WebviewViewIds;
+		timestamp: number;
+	},
+> extends GlElement {
+	@property({ type: String }) name!: string;
```

Comment:
Let's add `delegatesFocus` this since the App itself doesn't need to be focusable:

```typescript
	static override shadowRootOptions: ShadowRootInit = {
		...LitElement.shadowRootOptions,
		delegatesFocus: true,
	};
```

Reply:
Will do